java/java-impl/src/com/intellij/codeInsight/generation/OverrideImplementUtil.java

```diff
@@ -75,14 +75,33 @@ private OverrideImplementUtil() { }
    * @return list of method prototypes
    */
   @NotNull
-  public static List<PsiMethod> overrideOrImplementMethod(@NotNull PsiClass aClass, @NotNull PsiMethod method, boolean toCopyJavaDoc) throws IncorrectOperationException {
+  public static List<PsiMethod> overrideOrImplementMethod(@NotNull PsiClass aClass,
+                                                          @NotNull PsiMethod method,
+                                                          boolean toCopyJavaDoc) throws IncorrectOperationException {
+    return overrideOrImplementMethod(aClass, method, toCopyJavaDoc, false);
+  }
+
+  /**
+   * generate methods (with bodies) corresponding to given method declaration
+   *  there are maybe two method implementations for one declaration
+   * (e.g. EJB' create() -> ejbCreate(), ejbPostCreate() )
+   * @param aClass context for method implementations
+   * @param method method to override or implement
+   * @param toCopyJavaDoc true if copy JavaDoc from method declaration
+   * @param forceOverrideAnnotation use code style if false
```

Comment:
What happens in the language levels that don't require or prohibit override annotations?

Reply:
Currently, the caller would need to ensure that they only pass true if the language level supports override annotations. The calling code I wrote sets this flag to true if the user themselves has already typed out a `@java.lang.Override` annotation. The idea is that if the user writes `@Override` but the code style is set not to use `@Override`, the user still gets what they typed. If the language level doesn't support `@Override`, they probably wouldn't type it either (and or be able to access those completions)  I'm not sure what you mean by "if the language level doesn't require override annotations", isn't that always the case on Java >= 1.5?

Reply:
I meant Java <= 1.4. Having such a method in API could lead to misuse and bugs in such projects. Maybe the annotation should be added manually (if missing) in the caller instead, without any new API?

Reply:
That makes sense. I wouldn't know how to best go about that, though. What about using `PsiUtil.getLanguageLevel(aClass)` to check whether `@Override` is supported and silently ignoring `forceOverrideAnnotation` in case it isn't?

Reply:
That'd make a confusing API. I'd try avoiding adding new overloads here at all, of possible.